inc/Engine/CriticalPath/CriticalCSSSubscriber.php

```diff
@@ -437,10 +437,49 @@ public function insert_critical_css_buffer( $buffer ) {
 		$critical_css_content = str_replace( '\\', '\\\\', $critical_css_content );
 
 		$buffer = preg_replace( '#</title>#iU', '</title><style id="rocket-critical-css">' . wp_strip_all_tags( $critical_css_content ) . '</style>', $buffer, 1 );
+		$buffer = preg_replace( '#</body>#iU', $this->return_remove_cpcss_script() . '</body>', $buffer, 1 );
 
 		return $buffer;
 	}
 
+	/**
+	 * Returns JS script to remove the critical css style from frontend.
+	 *
+	 * @since 3.6
+	 *
+	 * @return string
+	 */
+	protected function return_remove_cpcss_script() {
+		if ( ! rocket_get_constant( 'SCRIPT_DEBUG' ) ) {
+			return '<script>const wprRemoveCPCSS = () => { document.getElementById( "rocket-critical-css" ).remove(); }; if ( window.addEventListener ) { window.addEventListener( "load", wprRemoveCPCSS ); } else if ( window.attachEvent ) { window.attachEvent( "onload", wprRemoveCPCSS ); }</script>';
+		}
+
+		return '
+			<script>
+				const wprRemoveCPCSS = () => {
+					document.getElementById( "rocket-critical-css" ).remove();
+				};
+				if ( window.addEventListener ) {
+					window.addEventListener( "load", wprRemoveCPCSS );
+				} else if ( window.attachEvent ) {
+					window.attachEvent( "onload", wprRemoveCPCSS );
+				}
+			</script>
+			';
+	}
+
+	/**
+	 * Adds wprRemoveCPCSS to excluded inline JS array.
+	 *
+	 * @since 3.6
+	 *
+	 * @param  array $excluded_inline Array of inline JS excluded from being combined.
+	 * @return array
+	 */
+	public function exclude_inline_js( $excluded_inline ) {
```

Comment:
Type hint to `array`.

```suggestion
	public function exclude_inline_js( array $excluded_inline ) {
```